From 57beb1e9d486ab2ecd565ea0845fd488da84adef Mon Sep 17 00:00:00 2001
From: jkreierh <jkreierh@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 16 Jan 2018 12:25:42 +0000
Subject: [PATCH 0819/1000] MANTIS 45801 massaiCfgXB*.exe version 2.0.1 -
 MANTIS 45801: MassaiCfgxB: many trace files - Re-design tracesFiler to:   -
 Not change files, just because a previous write failed   - Write an opening
 and closing entry for each file   - Create the a directory, if not exists

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1307 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h | 16 ++++++++--
 .../Massai/cpp/CfgXMgrB/inc/traceLocal.hpp    |  4 +--
 .../Massai/cpp/CfgXMgrB/src/CfgXMgrB.cpp      |  3 +-
 .../Massai/cpp/CfgXMgrB/src/massaiCfgXB.cpp   | 15 ++-------
 .../Massai/cpp/CfgXMgrB/src/traceLocal.cpp    | 31 +++++++------------
 5 files changed, 31 insertions(+), 38 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h b/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h
index a7677b2c..372c2001 100644
--- a/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h
+++ b/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h
@@ -514,8 +514,20 @@
 //! @par     Change:
 //! - MANTIS 47240: Config service meta data access
 //! - New services & functions.
-#define FILE_VER_BIN 2,0,0,0
-#define FILE_VER_DTS "2.0.0.0"
+// #define FILE_VER_BIN 2,0,0,0
+// #define FILE_VER_DTS "2.0.0.0"
+//
+//! @version 2.0.1
+//! @date    16.01.2018
+//! @author  JK
+//! @par     Change:
+//! - MANTIS 45801: MassaiCfgxB: many trace files
+//! - Re-design tracesFiler to:
+//!   - Not change files, just because a previous write failed
+//!   - Write an opening and closing entry for each file
+//!   - Create the a directory, if not exists
+#define FILE_VER_BIN 2,0,1,0
+#define FILE_VER_DTS "2.0.1.0"
 
 
 //-------------------------------------------------------------------------
diff --git a/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/traceLocal.hpp b/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/traceLocal.hpp
index bbcb168c..741437a8 100644
--- a/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/traceLocal.hpp
+++ b/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/traceLocal.hpp
@@ -7,7 +7,7 @@
  *
  * @author Juergen Kreierhoff
  *
- * Copyright (c) 2009-2015 MATERNA Information & Communications
+ * Copyright (c) 2009-2018 MATERNA Information & Communications
  *
  **************************************************************************
 **/
@@ -18,7 +18,7 @@
 #endif
 
 
-void traceLocal_ListenForChange();
+void traceLocal_Delete();
 
 void traceLocal_SetConsole();
 
diff --git a/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/CfgXMgrB.cpp b/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/CfgXMgrB.cpp
index 86e8c68e..2f339c01 100644
--- a/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/CfgXMgrB.cpp
+++ b/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/CfgXMgrB.cpp
@@ -7,7 +7,7 @@
  *
  * @author Juergen Kreierhoff
  *
- * Copyright (c) 2003-2017 MATERNA Information & Communications
+ * Copyright (c) 2003-2018 MATERNA Information & Communications
  *
  **************************************************************************
 **/
@@ -110,7 +110,6 @@ int main( int argc, char ** argv )
    // Traces
    //
    traceLocal_SetConsole();
-   traceLocal_ListenForChange();
    writeTrace
    (
       5, "", 0, "",
diff --git a/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/massaiCfgXB.cpp b/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/massaiCfgXB.cpp
index 3f86669c..d9619154 100644
--- a/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/massaiCfgXB.cpp
+++ b/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/massaiCfgXB.cpp
@@ -7,7 +7,7 @@
  *
  * @author Juergen Kreierhoff
  *
- * Copyright (c) 2003-2017 MATERNA Information & Communications
+ * Copyright (c) 2003-2018 MATERNA Information & Communications
  *
  **************************************************************************
 **/
@@ -146,13 +146,6 @@ int mDaemon_main( int argc, char const * const * argv )
    {
       traceLocal_SetPath( TraceDir );
    }
-   traceLocal_ListenForChange();
-
-   writeTrace
-   (
-      5, "", 0, "",
-      "________________________________________________________"
-   );
    writeTrace
    (
       5, "", 0, "",
@@ -251,11 +244,7 @@ int mDaemon_main( int argc, char const * const * argv )
    //
    // Exit program
    //
-   writeTrace
-   (
-      5, "", 0, "",
-      "massaiCfgXB - terminated."
-   );
+   traceLocal_Delete();
    return 0;
 }
 
diff --git a/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/traceLocal.cpp b/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/traceLocal.cpp
index 3ba3d492..1ef17651 100644
--- a/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/traceLocal.cpp
+++ b/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/traceLocal.cpp
@@ -7,7 +7,7 @@
  *
  * @author Juergen Kreierhoff
  *
- * Copyright (c) 2002-2015 MATERNA Information & Communications
+ * Copyright (c) 2002-2018 MATERNA Information & Communications
  *
  **************************************************************************
 **/
@@ -40,24 +40,22 @@ namespace
 
 
 
-void traceLocal_ListenForChange()
-{
-   // TODO:
-   // Perhaps with a config item, which is given as a start parameter.
-   // e.g. "-tracecfg:TRACES.CfgServive"
-}
-
-
-
-void traceLocal_SetConsole()
+void traceLocal_Delete()
 {
    if( l_writer )
    {
-      // stop previous writer
       tracesAdapter_set( 0 );
       // and delete it
       delete l_writer;
    }
+}
+
+
+
+void traceLocal_SetConsole()
+{
+   // Stop any previous writer
+   traceLocal_Delete();
    l_writer = new tracesToFile::writer( "con" );
    tracesAdapter_set( l_writer );
 }
@@ -66,13 +64,8 @@ void traceLocal_SetConsole()
 
 void traceLocal_SetPath( char const * path )
 {
-   if( l_writer )
-   {
-      // stop previous writer
-      tracesAdapter_set( 0 );
-      // and delete it
-      delete l_writer;
-   }
+   // Stop any previous writer
+   traceLocal_Delete();
    l_writer = new tracesToFileParts::writer( path, "CfgXMgrB" );
    tracesAdapter_set( l_writer );
 }
-- 
2.41.0.windows.1

